runtime/service_integ_test.go

```diff
@@ -533,3 +533,98 @@ vdf  254:80   0  512B  0 disk`
 			"context cancelled while waiting for container %s to exit, err: %v", containerName, ctx.Err())
 	}
 }
+
+func startAndWaitTask(ctx context.Context, t *testing.T, c containerd.Container) string {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	task, err := c.NewTask(ctx, cio.NewCreator(cio.WithStreams(nil, &stdout, &stderr)))
+	require.NoError(t, err, "failed to create task for container %s", c.ID())
+
+	exitCh, err := task.Wait(ctx)
+	require.NoError(t, err, "failed to wait on task for container %s", c.ID())
+
+	err = task.Start(ctx)
+	require.NoError(t, err, "failed to start task for container %s", c.ID())
+
+	select {
+	case exitStatus := <-exitCh:
+		require.NoError(t, exitStatus.Error(), "failed to retrieve exitStatus")
+		require.Equal(t, uint32(0), exitStatus.ExitCode())
+		require.Equal(t, "", stderr.String())
+	case <-ctx.Done():
+		require.Fail(t, "context cancelled",
+			"context cancelled while waiting for container %s to exit, err: %v", c.ID(), ctx.Err())
+	}
+
+	return stdout.String()
+}
+
+func TestRemoveDir_Isolated(t *testing.T) {
```

Comment:
It looks like this test is verifying the problem mentioned in #65 about IO not working when creating a new container no longer exists, which is good. However, it doesn't appear to currently be testing that all files we create related to the container are removed, e.g. the symlinks here that are parameterized by `containerID`: https://github.com/firecracker-microvm/firecracker-containerd/blob/e84a435e52acb36157536b5f3ddba090ed0da789/internal/vm/dir.go#L100-L132

I think it's worth adding some assertions to this test that those symlinks are deleted too.

Reply:
Yes. It is currently only testing the equivalent of `ctr run ... test`, `ctr c rm ... test` and `ctr run ... test`. Let me check the symlinks.

Reply:
The latest revision calls `os.Stat` to make sure that the associated directory of a container is being removed from the host OS correctly. Hard-coding the path is a bit annoying though.